src/libs/ReportUtils.js

```diff
@@ -1864,40 +1866,55 @@ function getModifiedExpenseMessage(reportAction) {
     if (hasModifiedCreated) {
         // Take only the YYYY-MM-DD value as the original date includes timestamp
         let formattedOldCreated = new Date(reportActionOriginalMessage.oldCreated);
         formattedOldCreated = format(formattedOldCreated, CONST.DATE.FNS_FORMAT_STRING);
-        return getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.created, formattedOldCreated, Localize.translateLocal('common.date'), false);
+        messageFragments.push(getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.created, formattedOldCreated, Localize.translateLocal('common.date'), false));
     }
 
     if (hasModifiedMerchant) {
-        return getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.merchant, reportActionOriginalMessage.oldMerchant, Localize.translateLocal('common.merchant'), true);
+        messageFragments.push(
+            getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.merchant, reportActionOriginalMessage.oldMerchant, Localize.translateLocal('common.merchant'), true),
+        );
     }
 
     const hasModifiedCategory = _.has(reportActionOriginalMessage, 'oldCategory') && _.has(reportActionOriginalMessage, 'category');
     if (hasModifiedCategory) {
-        return getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.category, reportActionOriginalMessage.oldCategory, Localize.translateLocal('common.category'), true);
+        messageFragments.push(
+            getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.category, reportActionOriginalMessage.oldCategory, Localize.translateLocal('common.category'), true),
+        );
     }
 
     const hasModifiedTag = _.has(reportActionOriginalMessage, 'oldTag') && _.has(reportActionOriginalMessage, 'tag');
     if (hasModifiedTag) {
-        return getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.tag, reportActionOriginalMessage.oldTag, Localize.translateLocal('common.tag'), true);
+        messageFragments.push(getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.tag, reportActionOriginalMessage.oldTag, Localize.translateLocal('common.tag'), true));
     }
 
     const hasModifiedBillable = _.has(reportActionOriginalMessage, 'oldBillable') && _.has(reportActionOriginalMessage, 'billable');
     if (hasModifiedBillable) {
-        return getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.billable, reportActionOriginalMessage.oldBillable, Localize.translateLocal('iou.request'), true);
+        messageFragments.push(
+            getProperSchemaForModifiedExpenseMessage(reportActionOriginalMessage.billable, reportActionOriginalMessage.oldBillable, Localize.translateLocal('iou.request'), true),
+        );
     }
+
+    return messageFragments.reduce((acc, value, index) => {
+        if (index === 0) {
+            return acc + value;
+        }
+        return `${acc}. ${value.charAt(0).toUpperCase()}${value.slice(1)}`;
```

Comment:
I think this looks a bit strange to have a mixture of upper/lower case and also a mix of punctuation:

![image](https://github.com/Expensify/App/assets/1228807/ccd32643-60bd-492c-b904-fa9a7515be8a)

The reason they don't start with a capital is because you should read it like "Cristi changed...". With the addition of these multiple fragments though, it looks off. Here is what I suggest:

1. Each message should start with a lower case
2. Each message should end with a period
3. If there are multiple fragments, then they should be like `changed the date to blah and the amount to blah.` (for two) and `changed the date to blah, the amount to blah, and the description to blah.` (for more than two)

Reply:
I implemented it as currently is in OldDot.
I really like your suggestion, I'll also ask in the wave4 channel if we want to change it in OldDot as well because it sounds more natural.